### DISCUSSION TITLE
Drawing Lag using Apple Pencil Pro

### DISCUSSION THREAD

**Discussion by leokvncrz** [keiwando/toonsquid#597] [General]
Comparing the performance on m1 ipad pro vs the latest m4 with apple pencil pro. 

I noticed significant delay specially on quick strokes. Sometimes quick lines doesnt register or have some delay. 

Apple pencil pro is working fine on native apps like apple notes and good notes.

**Comment by keiwando** [comment]
Thanks for reporting this. I have not been able to reproduce any drawing lag or other input delay with the M4 iPad Pro and Apple Pencil Pro, so this doesn't appear to be a consistent problem on that hardware.
Can you first try to manually [restart your iPad](https://support.apple.com/en-us/101603) and see if that fixes the issue. It certainly [wouldn't be the first time](https://github.com/keiwando/toonsquid/discussions/592#discussioncomment-9641422) that a random iPad issue fixed itself after a restart. If that didn't fix it, some more questions to maybe narrow down the cause:

Does this happen with all of the brushes or only some of them? 
Are you using vector brushes or pixel brushes?
Can you reproduce the delay in an empty project or only in an existing larger project?

**Reply by leokvncrz** [reply]
Thanks for your response. I tried restarting it but still showing lag even on empty project. 

I tried forgetting the apple pencil pro in bluetooth device and tried pairing it again. That seems to solve the lag issue. 

Thanks again. Glad to know it isn't an app issue.
